app/controllers/declaration/PreviousDocumentsRemoveController.scala

```diff
@@ -0,0 +1,90 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.declaration
+
+import controllers.actions.{AuthAction, JourneyAction}
+import controllers.navigation.Navigator
+import controllers.util.MultipleItemsHelper
+import forms.common.YesNoAnswer
+import forms.common.YesNoAnswer.YesNoAnswers
+import forms.declaration.{Document, PreviousDocumentsData}
+import javax.inject.Inject
+import models.{ExportsDeclaration, Mode}
+import models.requests.JourneyRequest
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.cache.ExportsCacheService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import utils.ListItem
+import views.html.declaration.previousDocuments.previous_documents_remove
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PreviousDocumentsRemoveController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  override val exportsCacheService: ExportsCacheService,
+  navigator: Navigator,
+  mcc: MessagesControllerComponents,
+  removePage: previous_documents_remove
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport with ModelCacheable {
+
+  def displayPage(mode: Mode, id: String): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
+    findDocument(id) match {
+      case Some(document) => Ok(removePage(mode, id, document, YesNoAnswer.form()))
+      case _              => returnToSummary(mode)
+    }
+  }
+
+  def submit(mode: Mode, id: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    findDocument(id) match {
+      case Some(document) =>
+        YesNoAnswer
+          .form()
+          .bindFromRequest()
+          .fold(
+            formWithErrors => Future.successful(BadRequest(removePage(mode, id, document, formWithErrors))),
+            validAnswer =>
+              validAnswer.answer match {
+                case YesNoAnswers.yes =>
+                  removeDocument(document).map { updatedDeclaration =>
+                    if (updatedDeclaration.flatMap(_.previousDocuments).map(_.documents).getOrElse(Seq.empty).nonEmpty) {
+                      returnToSummary(mode)
+                    } else {
+                      navigator.continueTo(mode, controllers.declaration.routes.PreviousDocumentsController.displayPage)
+                    }
```

Comment:
In `PreviousDocumentsSummaryController.displayPage` method there is a validation that will fall back to `PreviousDocumentsController` if there is no document in cache.
As I understand, you can therefore remove this `if` statement here and only redirect to `Summary`, bacause it will redirect again if needed.

Reply:
That's true, will fix this

Reply:
Done